jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MimeTypes.java

```diff
@@ -37,6 +40,40 @@
 public class MimeTypes
 {
     static final  Logger LOG = LoggerFactory.getLogger(MimeTypes.class);
+    private static final Index<Locale> KNOWN_LOCALES = new Index.Builder<Locale>()
+        .caseSensitive(false)
+        .withAll(() -> Arrays.stream(Locale.getAvailableLocales())
+            .collect(Collectors.toMap(Locale::toString, UnaryOperator.identity())))
+        .build();
+
+    /**
+     * @param languageDashCountry The language and optional country of the local. For example "en" or "en-GB".
+     * @return The {@link Locale} matching the string or null if a matching Locale is not known;
+     */
+    public static Locale getLocale(String languageDashCountry)
+    {
+        // First try for a direct hit in the known Locale cache
+        Locale locale = KNOWN_LOCALES.get(languageDashCountry);
+        if (locale != null)
+            return locale;
+
+        // failing that, create a new Locale instance to normalize the name
+        String language = HttpField.stripParameters(languageDashCountry);
+        int dash = languageDashCountry.indexOf('-');
+        if (dash < 0)
+        {
+            locale = new Locale(language);
+        }
+        else
+        {
+            language = languageDashCountry.substring(0, dash).trim();
+            String country = languageDashCountry.substring(dash + 1).trim();
+            locale = new Locale(language, country);
+        }
+
+        // look up normalized name in the known Locale cache.
+        return KNOWN_LOCALES.get(locale.toString());
```

Comment:
```suggestion
       Locale locale = Locale.forLanguageTag(languageDashCountry);
       return KNOWN_LOCALES.get(locale.toLanguageTag());
```

Reply:
@joakime yep, that's much much better.  We can even change the Index to just a Set